actionpack/lib/abstract_controller/base.rb

```diff
@@ -73,19 +73,28 @@ def internal_methods
       # * <tt>Set</tt> - A set of all methods that should be considered actions.
       def action_methods
         @action_methods ||= begin
+
           # All public instance methods of this class, including ancestors
           methods = (public_instance_methods(true) -
             # Except for public instance methods of Base and its ancestors
             internal_methods +
             # Be sure to include shadowed public instance methods of this class
-            public_instance_methods(false))
-
-          methods.map!(&:to_s)
+            public_instance_methods(false) -
+            # Except for flash methods added to this controller class
+            added_flash_types.to_a).map(&:to_s).uniq
```

Comment:
Do we need both of `uniq` here and `to_set` below?

Reply:
Well caught!